src/rviz/image/mouse_click.cpp

```diff
@@ -0,0 +1,127 @@
+#include "rviz/image/mouse_click.h"
+
+#include <ros/names.h>
+
+
+namespace rviz
+{
+
+MouseClick::MouseClick() : QObject()
+{
+}
+
+MouseClick::~MouseClick()
+{
+}
+
+
+void MouseClick::onInitialize(QObject* parent)
+{
+  setParent(parent);
+  parent->installEventFilter(this);
+
+  has_dimensions_ = false;
+  is_topic_name_ok_ = false;
+  node_handle_.reset(new ros::NodeHandle());
```

Comment:
This should inherit from Display's `update_nh_`. I suggest to pass the NodeHandle as an extra const-ref argument.

Reply:
Got it. I am passing the update_nh_ to the MouseClick constructor.